### PR TITLE
Coffee Editor NG

### PR DESCRIPTION
**Pull request by eneufeld**
This is a refactoring of the whole web page.

### CODE REVIEW

File: content/_index/coffeeeditoroverview/index.md [hunk 1]
```diff
@@ -3,11 +3,12 @@ fragment = "content"
...
 #title_align = "center" # Default is center, can be left, right or center
 +++
 <p style='text-align: center;'>
-An example web-based modeling tool using EMF.cloud technologies used for demonstration and documentation.
+The <em>Coffee Editor NG</em> is a comprehensive example modeling tool based on EMF.cloud technologies and can also act as an architecture blueprint for your custom modeling tool.
+This full-featured example tool is written entirely in Typescript and includes a central model hub serving a sample modeling language to a variety of editors, such as a diagram editor, a form-based editor, as well as a textual DSL editor.
 </p>
-<img src="images/coffeeeditordemo.gif" alt="Coffee Editor Overview" style="display: block; margin: auto;"/>
\ No newline at end of file
+<img src="images/coffeeeditordemo.gif" alt="Coffee Editor Overview" style="display: block; margin: auto;"/>
```
Comment (planger) [inline content/_index/coffeeeditoroverview/index.md:14]: Can we already provide an updated GIF or preferably a video that shows the NG. If possible in dark mode?

File: content/documentation/additionals/content.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+* <a target="_blank" href="https://eclipsesource.com/blogs/2021/02/25/the-emf-cloud-model-server/">The EMF Cloud Model Server</a>
+* <a target="_blank" href="https://eclipsesource.com/blogs/2021/02/03/how-to-build-a-tree-editor-in-eclipse-theia/">How to build a tree editor in Eclipse Theia</a>
+* <a target="_blank" href="https://eclipsesource.com/blogs/2020/07/24/a-web-based-modeling-tool-based-on-eclipse-theia/">A web-based modeling tool based on Eclipse Theia, EMF Cloud and GLSP</a>
+* <a target="_blank" href="https://eclipsesource.com/blogs/2019/12/16/introducing-emf-cloud/">Introducing EMF Cloud</a>
+* <a target="_blank" href="https://eclipsesource.com/blogs/2021/02/10/web-based-diagram-editor-features-in-eclipse-glsp/">Web-based diagram editor features in Eclipse GLSP</a>
+* <a target="_blank" href="https://eclipsesource.com/blogs/2021/01/27/glsp-diagrams-in-vs-code-theia-eclipse-and-plain-html/">GLSP: Diagrams in VS Code, Theia, Eclipse and plain HTML</a>
+
+## Videos
+
+* EclipseCon 2021: <a target="_blank" href="https://www.youtube.com/watch?v=mEr-v8t2ZQ8">Model validation, diffing and more with EMF.cloud</a>
```
Comment (planger) [inline content/documentation/additionals/content.md]:
```suggestion
* EclipseCon 2023: <a target="_blank" href="https://www.youtube.com/watch?v=U0TRt3aukTc">Building cloud-native (modeling) tools</a>
* EclipseCon 2021: <a target="_blank" href="https://www.youtube.com/watch?v=mEr-v8t2ZQ8">Model validation, diffing and more with EMF.cloud</a>
```

File: content/documentation/modelingLanguage/content.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
++++
+fragment = "content"
+title = "Defining a Modeling Language"
+weight = 160
+[sidebar]
+  sticky = true
++++
+
+In Model Hub, languages can be easily defined using Langium.
```
Comment (planger) [inline content/documentation/modelingLanguage/content.md]:
```suggestion
In Model Hub, languages can be easily defined using Eclipse Langium. 
```

File: content/documentation/langium/content.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+## Model Persistence
+
+A model persistence contribution provides language-specific methods to load and store models from and to a persistent storage.
+Based on the services generated by Langium, we can query the document storage from Langium and using the generic converter ensure that we return a serializable model for the Model Hub.
+Similarly, we can re-use the generated Langium infrastructure to store the model by converting the language model back to an AST model.
+Furthermore, we need to ensure that anytime a model is updated on the Langium side we properly update the model on the Model Hub side.
+We achieve that by installing a listener on the Langium side and using the Model Manager from the Model Hub to execute a PATCH command that updates the model in the Model Hub.
+
+For the Coffe Model, the persistence contribution may look something like this:
```
Comment (planger) [inline content/documentation/langium/content.md]:
```suggestion
For the Coffee Model, the persistence contribution may look something like this:
```

**Review by planger — CHANGES_REQUESTED** [review]
Very nice, thank you!

I think we should update the links here too:
image

1. In Browse the sources, we should either mention that the sources for NG are available soon and/or explicitly say that the sources linked here are pointing to the Java-based "product line". I prefer the first to avoid confusion, as this section is actually about NG.
2. In the Lean more section, I'd just point to [Maximilian's talk at EclipseCon](https://www.youtube.com/watch?v=U0TRt3aukTc) and the docs, but not to the blog post about the Java-based product line.

File: content/_index/coffeeeditoroverview/index.md (hunk 1 above)
Comment (eneufeld) [reply content/_index/coffeeeditoroverview/index.md:14]: it looks the same ;-).

**Review by planger — APPROVED** [review]
Thanks, looks good to me! :+1:

The only thing I noticed, is that there is a documentation page that still has todos on it:
https://deploy-preview-45--emfcloud.netlify.app/documentation/diagrameditor/

Imho this can be also done after merging, but we should not wait too long and prioritize completing this page to give the full picture.

Thank you!

**Comment by eneufeld** [comment]
Nina promised me an update tomorrow (21.12) morning

**Pull request merged**